### DISCUSSION TITLE
Cant stop loop

### DISCUSSION THREAD

**Discussion by mrracoonman** [Rapptz/discord.py#6043] [Q&A]
I'm trying to create a loop to set an announcement every so often and not have to worry about sending it every 30 min or so. I managed to get my bot to loop it, but I can't stop the loop.
This is what I have to create the loop, but what would I use to stop it?

@bot.command()
async def loop(ctx, arg):
    while True:
        await ctx.send(f'{arg}')
        await asyncio.sleep(1800)

@bot.command()
async def loopinfo(ctx, arg):
    bot.loop.create_task(loop(ctx, arg))

Win 10, python 3.8.6

**Comment by ShineyDev** [comment]
[`asyncio.loop.create_task`](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.create_task) doesn't just schedule a [`Task`](https://docs.python.org/3/library/asyncio-task.html#asyncio.Task) - it returns one. Set it to a variable and call its [`cancel`](https://docs.python.org/3/library/asyncio-task.html#asyncio.Task.cancel) method.

**Reply by Rapptz** [reply]
On top of this you probably don't want to make that function a command -- it's superfluous and allows someone to run this function through Discord.

**Reply by mrracoonman** [reply]
Is there any way for you to provide an example so I can better understand what you mean?
